### ISSUE TITLE
[Footer]: QA: Select geographic area modal Screen reader focus issue

### ISSUE THREAD

**Issue by Gopi916** [carbon-design-system/carbon-for-ibm-dotcom#8856]
### Description

Page Element:
Select geographic area modal

Issue:
1. Initially When the modal dialog is displaied the screen reader focus is trapped within the modal. However when the user switches to another window and navigate back to the modal, the focus is no longer trapped within the modal. The parent screen is exposed .
2. When the focus is not trapped, there are couple of buttons that are identified by the screen reader just before and after the dialog with the label "start" and "end".

Expected:
1. Here the screen reader focus is expected to be trapped within the modal even after the user switches to another window and navigates back to the modal
2. The visually hidden buttons should be hidden from SR as well.

User Impact:
When the screen reader focus is not restricted within the modal dialog, it also identifies parent page content that disorients screen reader users.

### Component(s) impacted

Footer (Default)

### Browser

Firefox

### Carbon for IBM.com version

v1.34.0-rc.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://www.ibm.com/standards/carbon/web-components/?path=/story/

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
